### PR TITLE
Add DialogProvider and ConfirmationDialog

### LINKED ISSUE #49: [FEATURE] Get user's confirmation before creating a user revision copy
These models can be very large in size, it'd be a good idea to let users confirm this action.

A sub-task to this might be creating a confirmation modal.

### LINKED ISSUE #52: [FEATURE] Add modal dialog component
Add a component that shows a modal dialog.

- Block whole page (CSS overlay)
- Canceled when clicking on either `X`, overlay or `Cancel` action button
- Accepts callbacks for `onCancel` and `onConfirm`
- Property for name of confirmation action
- Properties for title and text

https://mui.com/material-ui/react-dialog/

### PR DESCRIPTION
**Pull request by mferrera**
This PR creates a context in which confirmation dialogs can be issued. It offers a way to customize the text displayed on the dialog and buttons, as well as executing callbacks passed to the Confirm and Cancel buttons.

Resolves #49 
Resolves #52

### CODE REVIEW

File: webpack.config.renderer.js [hunk 1]
```diff
@@ -338,7 +338,7 @@ module.exports = () => {
         },
         resolve: {
             plugins: [new TsconfigPathsPlugin()],
-            extensions: [".tsx", ".ts", ".js"],
+            extensions: [".tsx", ".ts", ".js", ".mjs"],
```
Comment (rubenthoms) [inline webpack.config.renderer.js:341]: Is this related to this PR or just in general?

**Review by rubenthoms — APPROVED** [review]
Nice UX. As we are only going to have one dialog open at once, I think the provider approach can work. We could also extend it with other dialog types as we see need (I guess that was your intention).

File: webpack.config.renderer.js (hunk 1 above)
Comment (mferrera) [reply webpack.config.renderer.js]: I may have had a good reason for adding this but I don't remember it now 😄 removed since it doesn't seem relevant to the PR
